src/plugins/irc-events/link.js

```diff
@@ -64,7 +64,7 @@ function parse(msg, url, res, client) {
 	case "image/gif":
 	case "image/jpg":
 	case "image/jpeg":
-		if (res.size < (Helper.config.prefetchMaxImageSize * 1024)) {
+		if (res.text.length < (Helper.config.prefetchMaxImageSize * 1024)) {
```

Comment:
`text.length` would be whatever fetched content length, so it will be there, yes. What we can do is check if `content-length` is under `prefetchMaxImageSize` and compare it to `text.length` to see if it's true. If it's not, then do not send the image to client.